app/services/name_synchronizer.rb

```diff
@@ -15,10 +15,12 @@ def initialize(object)
   # we have to use write_attribute, since host#name= is delegated to primary interface
   # which triggers the sync
   def sync_name
-    @host.send :write_attribute, :name, @interface.name
+    name = Setting[:append_domain_name_for_hosts] ? @interface.name : @interface.shortname
+    @host.send :write_attribute, :name, name
   end
 
   def sync_required?
-    @interface.primary? && @host.present? && (@host.name != @interface.name)
+    name = Setting[:append_domain_name_for_hosts] ? @interface.name : @interface.shortname
```

Comment:
please extract this to some private method so it's not duplicated in `sync_name` and `sync_required?`